src/middlewared/middlewared/schema.py

```diff
@@ -81,6 +81,9 @@ def clean(self, value):
                 raise Error(self.name, 'attribute required')
         return value
 
+    def has_private(self):
```

Comment:
Can this be named `is_private` and just be used in `dump` like `if self.is_private(): return "***"`?

Reply:
So `is_private` as a name is quite similar to `private` which means the attribute in question is private. However, `has_private` serves a use case which indicates that the attribute has other children which are private and if the attribute is being used somewhere, the user of the attribute can decide how to handle the children case

Reply:
If anything else is private, children are also private, how else can it be?

Reply:
How about attribute is not private but it has a few children which are, we do not want to hide the parent then, do we ?

Reply:
When attribute is not private, we dump it, and children will hide themselves when dumping recursively. I've mean use it like this in `List.dump`